### PR TITLE
Add wasd camera plugin

### PR DESCRIPTION
**Pull request by Adam-**

### CODE REVIEW

**Comment by devLotto** [comment]
I just came to think of French keyboards (AZERTY), as they have ZQSD in place of WASD, and DVORAK keyboards that has <AOE. Maybe Keybind should be used with WASD as default?

File: runelite-client/src/main/java/net/runelite/client/plugins/wasdcamera/WASDCameraPlugin.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	@Inject
+	private ClientThread clientThread;
+
+	@Inject
+	private KeyManager keyManager;
+
+	@Inject
+	private WASDCameraListener inputListener;
+
+	boolean typing;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/wasdcamera/WASDCameraPlugin.java]: private and `@Getter`

File: runelite-client/src/main/java/net/runelite/client/plugins/wasdcamera/WASDCameraPlugin.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	{
+		clientThread.invoke(() ->
+		{
+			if (client.getGameState() == GameState.LOGGED_IN)
+			{
+				unlockChat();
+			}
+		});
+
+		keyManager.unregisterKeyListener(inputListener);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/wasdcamera/WASDCameraPlugin.java]: reset `typing` to false

File: runelite-client/src/main/java/net/runelite/client/plugins/wasdcamera/WASDCameraListener.java [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+	public void keyPressed(KeyEvent e)
+	{
+		if (!plugin.chatboxFocused())
+		{
+			return;
+		}
+
+		if (!plugin.typing)
+		{
+			if (config.up().matches(e))
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/wasdcamera/WASDCameraListener.java]: Most of this method is duplicated in keyReleased and keyTyped, extract this to common method and use it in all listeners

**Pull request merged**